### PR TITLE
dev-libs/dmalloc: redo patches

### PR DESCRIPTION
**Pull request by parona-source**
* Split changes, drop redundant and problematic changes.

Closes: https://bugs.gentoo.org/920106

### CODE REVIEW

File: dev-libs/dmalloc/dmalloc-5.6.5-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+src_test() {
+	# mv: cannot stat 'aout': No such file or directory
+	emake -j1 heavy
+}
+
+src_install() {
+	default
+
+	soname_link() {
```
Comment (thesamesam) [inline dev-libs/dmalloc/dmalloc-5.6.5-r1.ebuild:48]: Double check the order is right wrt f67f7c6453fdd128489ea675935302099d66c1f5?

Comment (parona-source) [reply dev-libs/dmalloc/dmalloc-5.6.5-r1.ebuild:48]: Didn't know about this and was just trying to keep as close to original behaviour as possible.

Following that order would simplify the patches.

Comment (parona-source) [reply dev-libs/dmalloc/dmalloc-5.6.5-r1.ebuild:48]: Never mind, got the symbolic link directions the wrong around. It wont simplify it.

But the order is still wrong in the ebuild.

Comment (parona-source) [reply dev-libs/dmalloc/dmalloc-5.6.5-r1.ebuild:48]:
```
$ qlist dmalloc -v
dev-libs/dmalloc-5.6.5-r1 CONTENTS:
/usr/share/doc/dmalloc-5.6.5-r1/TODO.bz2
/usr/share/doc/dmalloc-5.6.5-r1/README.md.bz2
/usr/bin/dmalloc
/usr/lib64/libdmallocxx.so -> libdmallocxx.so.5
/usr/lib64/libdmallocxx.so.5 -> libdmallocxx.so.5.6.5
/usr/lib64/libdmalloc.so -> libdmalloc.so.5
/usr/lib64/libdmalloc.so.5 -> libdmalloc.so.5.6.5
/usr/lib64/libdmalloc.so.5.6.5
/usr/lib64/libdmallocthcxx.so.5.6.5
/usr/lib64/libdmallocth.so.5.6.5
/usr/lib64/libdmallocxx.so.5.6.5
/usr/include/dmalloc.h
/usr/lib64/libdmallocth.so.5 -> libdmallocth.so.5.6.5
/usr/lib64/libdmallocth.so -> libdmallocth.so.5
/usr/lib64/libdmallocthcxx.so.5 -> libdmallocthcxx.so.5.6.5
/usr/lib64/libdmallocthcxx.so -> libdmallocthcxx.so.5
```

**Comment by thesamesam** [comment]
Thank you!

**Pull request closed without merging**